### PR TITLE
feat: implement `source`

### LINKED ISSUE #94: Implement "source a.sh"
So that I can add my local configuration and source it.

As a workaround, I do `cat .shellrc` and copy & paste it manually.

### PR DESCRIPTION
**Pull request by wolfv**
Fixes #94.

I am not sure about the exact semantics. Currently, everything that is modified in the sourced' script is also going to be modified in the parent script.

### CODE REVIEW

**Comment by certik** [comment]
Probably time to fix https://github.com/prefix-dev/shell/issues/82. :) And define the exact semantics with tests, and we can always check that Bash behaves the same way.

I think the current semantics is correct, I think sourcing a script is equivalent to copy & pasting the script into the parent script and execute there.

**Comment by certik** [comment]
I tested this on Windows and it works! I can now load `~/.shellrc` to load paths for pixi and my `~/bin` directory.

I think this unblocks me, I can now create scripts for various conda environments, and this might be all that was needed to start using shell for my daily work.

**Comment by certik** [comment]
Found a bug: https://github.com/prefix-dev/shell/issues/113.

File: crates/shell/src/execute.rs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+};
+
+pub async fn execute_inner(text: &str, state: ShellState) -> anyhow::Result<ExecuteResult> {
+    let list = deno_task_shell::parser::parse(text);
+
+    let stderr = ShellPipeWriter::stderr();
+    let stdout = ShellPipeWriter::stdout();
+    let stdin = ShellPipeReader::stdin();
+
+    if let Err(e) = list {
```
Comment (prsabahrami) [inline crates/shell/src/execute.rs:14]: Is there any reason we are bailing on syntax errors? This causes shell to exit on syntax errors in interactive mode. 
Previously the error was being written to `stderr`.

Comment (wolfv) [reply crates/shell/src/execute.rs:14]: I think we might want to catch the error somewhere upstream!

**Pull request merged**